File: Advent of Code 2025/AC25D1.py
total = 50
rotations = 0

with open("D1input.txt", "r") as f:
    for lines in f:
        rotations += int(lines[1:])//100
        remainder = int(lines[1:]) % 100
        if lines[0] == "L":
            if total + remainder >= 100:
                rotations += 1
            total = (total +remainder) % 100
        elif lines[0] == "R":
            if total > 0 and total - remainder <= 0:
                rotations +=1

            total = (total - remainder + 100)%100

print(rotations)

File: Advent of Code 2025/AC25D1.rs
use std::fs::File;
use std::io::{self, BufRead, BufReader};


fn main() -> Result<(), Box<dyn std::error::Error>>{
    let file = File::open("R27.txt")?;
    let reader = BufReader::new(file);
    let mut total = 0;

    for line in reader.lines() {
        let line = line?; // Handle potential errors for each line
        line
        println!("{}", line);
    }
    Ok(())
    
}